autode/conformers/conformers.py

```diff
@@ -234,6 +234,67 @@ def prune_diff_graph(
         logger.info(f"Pruned on connectivity {n_prev_confs} -> {len(self)}")
         return None
 
+    def _set_boltzmann_weight(self, temp: float) -> None:
+        """Sets self._boltzmann_weight for each conformer"""
+
+        min_energy = self.lowest_energy.energy.to("J")
+        k_b = 1.380649 * 10**-23  # Units: J/K
+
+        for conf in self:
+
+            normalised_energy = conf.energy.to("J") - min_energy
+            conf._boltzmann_weight = np.exp(-normalised_energy / (temp * k_b))
+
+        return None
+
+    def prune_with_boltzmann(
+        self, temp: float = 298.15, threshold: float = 0.9
+    ) -> None:
+        """
+        Keep the lowest energy conformers whose cumulative weights are up to
+        the threshold (default 90%). Conformer energies must be in Ha.
+        -----------------------------------------------------------------------
+
+        Arguments:
+            threshold (float): Cumlative weight threshold.
+            temp (float): Temperature in K.
+        """
+        assert threshold <= 1.0, "The treshold must be below 1.0"
+
+        self._set_boltzmann_weight(temp)
+
+        normalisation_constant = sum(conf._boltzmann_weight for conf in self)
+
+        # Sort conformers in ascending order by Boltzmann weight
+        self.sort(key=lambda conf: conf.boltzmann_weight)
+
+        cumulative_weight = 0
+
+        for idx, conf in enumerate(self):
+
+            conf._boltzmann_weight = (
+                conf._boltzmann_weight / normalisation_constant
+            )
```

Comment:
I'd expect this to be set in the `_set_boltzmann_weight` function? think `Conformer.boltzmann_weight` should be _either_ the normalised or unnormalised weight